apps/wallet/src/ui/app/helpers/sortAccounts.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { type AccountType, type SerializedUIAccount } from '_src/background/accounts/Account';
+import { isLedgerAccountSerializedUI } from '_src/background/accounts/LedgerAccount';
+import { isMnemonicSerializedUiAccount } from '_src/background/accounts/MnemonicAccount';
+import { isZkAccountSerializedUI } from '_src/background/accounts/zk/ZkAccount';
+
+const sortFns: Partial<
+	Record<AccountType, (a: SerializedUIAccount, b: SerializedUIAccount) => number>
+> = {
+	ledger: (a, b) =>
+		isLedgerAccountSerializedUI(a) && isLedgerAccountSerializedUI(b)
+			? a.derivationPath.localeCompare(b.derivationPath)
+			: 0,
+	'mnemonic-derived': (a, b) =>
+		isMnemonicSerializedUiAccount(a) && isMnemonicSerializedUiAccount(b)
+			? a.derivationPath.localeCompare(b.derivationPath)
+			: 0,
+};
+
+function getKey(account: SerializedUIAccount): string {
+	if (isMnemonicSerializedUiAccount(account)) return account.sourceID;
+	if (isZkAccountSerializedUI(account)) return account.provider;
+	return account.type;
+}
+
+export const defaultSortOrder: AccountType[] = [
+	'zk',
+	'mnemonic-derived',
+	'imported',
+	'ledger',
+	'qredo',
+];
+
+export function groupByType(accounts: SerializedUIAccount[]) {
```

Comment:
If I read this right, we are not sorting the groups within the types right now. We rely on the initial order of the accounts and on how browsers iterate on the object's fields

lets say we have 2 mnemonics source1 and source2 and 2 accounts for each
The final result will be something like
```{ 'mnemonic-derived: { source1: [derPath1, derPath2], source2: [derPath1, derPath2]}```

but also 
```{ 'mnemonic-derived: { source2: [derPath1, derPath2], source1: [derPath1, derPath2]}```
depending the order of the initial accounts

This probably would benefit from some unit tests! 😄